climada/hazard/tc_tracks.py

```diff
@@ -799,6 +805,56 @@ def read_netcdf(self, folder_name):
             track.attrs['orig_event_flag'] = bool(track.orig_event_flag)
             self.data.append(track)
 
+    def to_geodataframe(self, as_points=False, joint=True):
+        """Transform this TCTracks instance into a GeoDataFrame.
+
+        Parameters:
+            as_points (bool): If true, construct LineString or single Point
+                geometries. A point is created if a track has length one.
+                If set to false, the geometries are returned as points with an
+                additional timestamp column.
+            joint (bool): If using as_points, return geometries joined to
+                metadata. If set to False, return a tuple of two GeoDataFrames
+                that can be joined on their index.
+
+        Returns:
+            GeoDataFrame
+        """
+        gdf = gpd.GeoDataFrame(
+            [dict(track.attrs) for track in self.data]
+        )
+
+        if as_points:
+            times = np.concatenate([track.time.data for track in self.data])
+            points = np.concatenate([
+                np.c_[track.lon, track.lat] for track in self.data
+            ])
+
+            # repeat each idx according to the number of timesteps
+            lens = [track.time.size for track in self.data]
+            idx = np.repeat(gdf.index.to_numpy(), lens)
+
+            gdf_long = gpd.GeoDataFrame({
+                'idx': idx,
+                'time': times,
+                'geometry': [Point(p[0], p[1]) for p in points],
+            }).set_index('idx')
+
+            if joint:
+                gdf = gdf.join(gdf_long)
+            else:
+                return gdf, gdf_long
```

Comment:
Unless there is a very good reason not to, the return value of a method/function should have a single (predefined) type - with the exception of `None` as the alternative, generic no-return type.\
In this particular case it's preferable to define two methods, one calling the other with a gdf and a tuple of gdfs respectively as return values.\
But consider whether it is not even preferable to leave the joining or splitting up again to the caller.

Reply:
I did it like this because "unjoining" would be inefficient on large tracksets. Given that returning the tuple is possible with no overhead, I feel that this is acceptable.

Reply:
Inefficient in terms of performance or in terms of convenience of the caller?

Reply:
In terms of performance. See my post below, I presumed wrongly 😀.